### PR TITLE
Follow google auth

### PR DESCRIPTION
**Pull request by Nek-**
This should lead to release 0.2.0

### CODE REVIEW

File: lib/Nekland/YoutubeApi/Http/Auth/JsonFileServiceAuth.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    protected function getIss()
+    {
+        $this->loadFile();
+
+        return $this->json['client_email'];
+    }
+
+    /**
+     * @return string
+     * @throws \Nekland\YoutubeApi\Exception\AuthException
```
Comment (mcsky) [inline lib/Nekland/YoutubeApi/Http/Auth/JsonFileServiceAuth.php]: This method don't throw an exception

File: lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php [hunk 2]
```diff
@@ -109,7 +125,7 @@ private function getPrivateKey()
      * @return mixed
      * @throws \Nekland\YoutubeApi\Exception\MissingOptionException
      */
-    private function getOption($option, $default=null)
+    protected function getOption($option, $default=null)
```
Comment (mcsky) [inline lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php]: Isn't redefined in subclass, the protected is not needed IMHO

File: test/Nekland/Tests/Http/Auth/AuthServiceTrait.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        $request->send(Argument::any())->shouldBeCalled()->willReturn($response->reveal());
+        $client->post(Argument::cetera())->willReturn($request->reveal());
+
+        return $client->reveal();
+    }
+
+    protected function getGoogleResponse()
+    {
+        return '{
+    "access_token" : "ya29.1.AADHN_WnGjqYiAcnONRLFDOfXia5XZLFO4RSyEhWtQPAvYpgPYiQj89c7UsrAs5_",
```
Comment (mcsky) [inline test/Nekland/Tests/Http/Auth/AuthServiceTrait.php]: indentation

File: test/Nekland/Tests/Http/Auth/JsonFileServiceAuthTest.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        // This file is a fake, of course.
+        $file = __DIR__ . '/../../../../test_data/given_by_google.json';
+
+        $auth->setOptions(['json_file' => $file]);
+        $request = new Request('POST', 'http://fake.com');
+        $auth->auth($request);
+
+        $this->assertContains('ya29.1.AADHN_WnGjqYiAcnONRLFDOfXia5XZLFO4RSyEhWtQPAvYpgPYiQj89c7UsrAs5_',(string)$request->getHeader('Authorization'));
+    }
+}
```
Comment (mcsky) [inline test/Nekland/Tests/Http/Auth/JsonFileServiceAuthTest.php]: you could add a test if the `JsonFileServiceAuth` is used without the `json_file` option the process should throw an exception

**Review by mcsky — CHANGES_REQUESTED** [review]

File: lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php (hunk 2 above)
Comment (Nek-) [reply lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php]: It is used !

File: test/Nekland/Tests/Http/Auth/AuthServiceTrait.php (hunk 3 above)
Comment (Nek-) [reply test/Nekland/Tests/Http/Auth/AuthServiceTrait.php]: The indentation would be bad if I add a tab!

File: lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php (hunk 2 above)
Comment (Awkan) [reply lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php]: Only [used](https://github.com/Nekland/YoutubeApi/pull/13/files#diff-e3e73917769c2433b3b97c9841b84792R58), not redefined

File: test/Nekland/Tests/Http/Auth/JsonFileServiceAuthTest.php (hunk 4 above)
Comment (Nek-) [reply test/Nekland/Tests/Http/Auth/JsonFileServiceAuthTest.php]: Indeed 😄 . Done!

File: lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php (hunk 2 above)
Comment (Nek-) [reply lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php]: it still needs to be protected :)

**Comment by Nek-** [comment]
@mcsky thanks for feedbacks!

File: lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php (hunk 2 above)
Comment (Awkan) [reply lib/Nekland/YoutubeApi/Http/Auth/ServiceAuth.php]: My bad 👍

**Pull request merged**